lib/OntologyAPI/OntologyAPIImpl.py

```diff
@@ -0,0 +1,606 @@
+           (defaults to 20) offset - optional - number of results to skip
+           (defaults to 0)) -> structure: parameter "id" of type "ID"
+           (Ontology term id, such as "GO:0000002"), parameter "ts" of Long,
+           parameter "ns" of String, parameter "limit" of Long, parameter
+           "offset" of Long
+        :returns: instance of type "GenericResults" (Generic results stats -
+           Query execution information from ArangoDB. results - array of
+           objects of results. ts - Timestamp used in the request ns -
+           Ontology namespace used in the request.) -> structure: parameter
+           "stats" of unspecified object, parameter "results" of list of
+           unspecified object, parameter "ts" of Long, parameter "ns" of
+           String
+        """
+        # ctx is the context object
+        # return variables are: returnVal
+        #BEGIN get_descendants
+        validated_params=self.validate_params(GenericParams);
+        results = re_api.query("get_descendants", validated_params)
+
+        returnVal={"stats": results["stats"], "results": results["results"], "ts": validated_params["ts"], "ns": validated_params["ns"]}
+        #END get_descendants
+
+        # At some point might do deeper type checking...
+        if not isinstance(returnVal, dict):
+            raise ValueError('Method get_descendants return value ' +
+                             'returnVal is not type dict as required.')
+        # return the results
+        return [returnVal]
+
+    def get_ancestors(self, ctx, GenericParams):
+        """
+        Retrieve ancestors of an ontology term by ID
+        :param GenericParams: instance of type "GenericParams" (Generic
+           Parameters id - required - ontology term id, such as "GO:0016209"
+           ts - optional - fetch documents with this active timestamp,
+           defaults to now ns - optional - ontology namespace to use,
+           defaults to "go" limit - optional - number of results to return
+           (defaults to 20) offset - optional - number of results to skip
+           (defaults to 0)) -> structure: parameter "id" of type "ID"
+           (Ontology term id, such as "GO:0000002"), parameter "ts" of Long,
+           parameter "ns" of String, parameter "limit" of Long, parameter
+           "offset" of Long
+        :returns: instance of type "GenericResults" (Generic results stats -
+           Query execution information from ArangoDB. results - array of
+           objects of results. ts - Timestamp used in the request ns -
+           Ontology namespace used in the request.) -> structure: parameter
+           "stats" of unspecified object, parameter "results" of list of
+           unspecified object, parameter "ts" of Long, parameter "ns" of
+           String
+        """
+        # ctx is the context object
+        # return variables are: returnVal
+        #BEGIN get_ancestors
+        validated_params=self.validate_params(GenericParams);
```

Comment:
Same comments from the get_descendants method can apply to all the other methods